src/view/ViewCommandHandlers.js

```diff
@@ -266,6 +266,23 @@ define(function (require, exports, module) {
         }
     }
 
+
+    /**
+     * Font smoothing setter to set the anti-aliasing type for the code area on Mac.
+     * @param {string} aaType The antialiasing type to be set.It can take either "subpixel-antialiased" or "antialiased"
+     */
+    function setMacFontSmoothingType(aaType) {
+        var $editor_holder  = $("#editor-holder");
+
+        // Add/Remove the class based on the preference. Also
+        // default to subpixel AA in case of invalid entries.
+        if (aaType === "antialiased") {
+            $editor_holder.removeClass("subpixel-aa");
+        } else if (aaType === "subpixel-antialiased" || aaType !== "antialiased") {
```

Comment:
I could change the code not to react to a bad preference but that might lead to unexpected behavior. Let's suppose the current value is `antialiased`, which makes the code editor render using gray scale AA. Say the preference is now changed to `zantialiased`, an invalid entry. At this point of time, if we don't change  AA to default because of the bad preference, on the next launch of Brackets the user would see a rendering different than the one in the previous session. IMO we should default to subpixel AA in case of bad preference.About moving this to a switch statement, that seems to be a good idea and I will move it to a switch statement.